flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/streamoperator/StreamOperator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.streamoperator;
+
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.ml.common.AlgoOperator;
+import org.apache.flink.ml.streamoperator.source.TableSourceStreamOp;
+import org.apache.flink.table.api.Table;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Base class of streaming algorithm operators.
+ */
+public abstract class StreamOperator<T extends StreamOperator <T>> extends AlgoOperator<T> {
```

Comment:
The name `StreamOperator` is colliding with `org.apache.flink.streaming.api.operators.StreamOperator` which is a widely established class in Flink. Can we change it to something like `StreamAlgoOperator`? We may need to change the `BatchOperator` to `BatchAlgoOperator` as well.

Reply:
Thanks for your advice and offline discussion. In our ML algorithm implementations and applications, we need not use org.apache.flink.streaming.api.operators.StreamOperator, and we'd like to use BatchOperator in ML lib.